static/modules/gamemanager.js

```diff
@@ -123,6 +130,16 @@ class GameManager {
       }
     });
 
+    document.addEventListener("keydown", e => {
+      const code = e.keyCode;
+      const key = String.fromCharCode(code);
+      // press P to pause
+      if (key == "P") {
+        toggleGuiElement("pausescreen");
+        this.gamePause = !this.gamePause;
+      }
+    });
+
```

Comment:
Do we need a document listener or could we handle this like how we handle a "shoot" input? Maybe we could have a persistent, invisible controller entity that always exists and handles inputs like pause (and also fullscreen could get moved to there too)

Reply:
does this work for lowercase p?

Reply:
keyCode is actually deprecated, we should use (KeyboardEvent.key)[https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/key] instead. But this will be removed anyway.